pkg/transformers/refvars.go

```diff
@@ -41,49 +43,84 @@ func NewRefVarTransformer(
 	}
 }
 
+// replacePrimitiveType checks if the field is a string. If not it will return
+// the field. In case it is a string, it will expand the field, and perform
+// a deepCopy of the expanded value in case it is not of primitiv type.
```

Comment:
// replaceVarsIfStringField checks if the incoming field contains a string.
// If so, the field is processed to replace variables.
// If not, the field is returned as is.

This names the function like a trailing conditional, e.g. http://jerodsanto.net/2015/03/trailing-conditionals-considered-harmful/

Reply:
Thank you for the advise. Renamed method to replaceStringField